storey/steps/sample.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime, timezone
+from typing import Optional, Union
+
+from storey import Filter, Event
+
+
+class Sample(Filter):
+    """Samples stream events, passes a single event downstream after either reaching 'rate_count' or 'rate_seconds'"""
+
+    def __init__(
+        self,
+        rate_count: Optional[int] = None,
+        rate_seconds: Optional[Union[int, float]] = None,
```

Comment:
It's not really *rate* isn't it? more like `max_events` and `timeout` or `emit_after_seconds`
Also add docs for theses parameters